components/RevisionCard.vue

```diff
@@ -518,6 +613,52 @@
         }
       });
 
+      // Author-based analysis
+      var curRevisionInfoAuthor: CrossEditSuspiciousPatternsInfo = {
+        mode: "author",
+        url: "https://en.wikipedia.org/w/api.php?origin=*",
+        windowSize: 10,
+        baseline: 0.0,
+        percentage: 0.0,
+        margin: 0.1,
+        warningTimeframe: this.warningTimeframeAuthor,
+        warningThreshold: this.warningThresholdAuthor,
+        revID: this.wikiRevId,
+        axiosClient: this.$axios,
+      }
+      this.CrossEditSuspiciousPatternsInstanceAuthor = new CrossEditSuspiciousPatterns(curRevisionInfoAuthor);
+      var decisionInfoAuthor = await this.CrossEditSuspiciousPatternsInstanceAuthor.analyze();
+      if (decisionInfoAuthor.type != "") {
+        this.displayChoiceAuthor = true;
+        this.displayHistoryAuthor = true;
+        this.choiceInfoAuthor = decisionInfoAuthor;
+        this.previousRevisionInfosAuthor = decisionInfoAuthor.previousRevisionInfos;
+      }
+      console.log(this.previousRevisionInfosAuthor);
+
+      // Article-based analysis
+      var curRevisionInfoArticle: CrossEditSuspiciousPatternsInfo = {
+        mode: "article",
+        url: "https://en.wikipedia.org/w/api.php?origin=*",
+        windowSize: 10,
+        baseline: 0.0,
+        percentage: 0.0,
+        margin: 0.1,
+        warningTimeframe: this.warningTimeframeArticle,
+        warningThreshold: this.warningThresholdArticle,
+        revID: this.wikiRevId,
+        axiosClient: this.$axios,
+      }
+      this.CrossEditSuspiciousPatternsInstanceArticle = new CrossEditSuspiciousPatterns(curRevisionInfoArticle);
+      var decisionInfoArticle = await this.CrossEditSuspiciousPatternsInstanceArticle.analyze();
+      if (decisionInfoArticle.type != "") {
```

Comment:
What does it mean by if "type !='' " ? This looks confusing, can you add comment?